### PR TITLE
Turn off auto linting by default, and on for swe_bench

### PR DESCRIPTION
**Pull request by li-boxuan**
This is built on top of #1794. Linting after editing now is turned off by default, but can be enabled via config, e.g.

```bash
ENABLE_AUTO_LINT=true poetry run python opendevin/runtime/docker/ssh_box.py
```

SWE-bench is not impacted. ENABLE_AUTO_LINT is turned ON for it.

### CODE REVIEW

File: evaluation/swe_bench/swe_env_box.py [hunk 1]
```diff
@@ -73,6 +73,10 @@ def get_box_for_instance(
             )
         config.workspace_base = workspace_mount_path
         config.workspace_mount_path = workspace_mount_path
+
+        # linting after editing boosts benchmark scores
+        config.enable_auto_lint = True
```
Comment (xingyaoww) [inline evaluation/swe_bench/swe_env_box.py:78]: Maybe you can update the `config.toml` example in the `README.md` of `evaluation/swe_benc` - just for safe?

**Review by xingyaoww — APPROVED** [review]
Overall LGTM!

File: evaluation/swe_bench/swe_env_box.py (hunk 1 above)
Comment (li-boxuan) [reply evaluation/swe_bench/swe_env_box.py:78]: make sense. Updated.

File: evaluation/swe_bench/swe_env_box.py [hunk 2]
```diff
@@ -73,6 +73,10 @@ def get_box_for_instance(
             )
         config.workspace_base = workspace_mount_path
         config.workspace_mount_path = workspace_mount_path
+
+        # linting after editing boosts benchmark scores
```
Comment (li-boxuan) [inline evaluation/swe_bench/swe_env_box.py:77]:
```suggestion
        # linting python after editing helps LLM fix indentations
```

**Pull request merged**